app/models/manageiq/providers/google/cloud_manager/event_catcher/event_producer.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::EventProducer
+  def initialize(ems)
+    @ems = ems
+  end
+
+  # Poll for messages. This makes a service call to the Google Pubsub service
+  # and returns any messages generated from the activity log. Note that these
+  # messages are acknowledged before returning, so the caller has no
+  # responsibility to explicitly acknowledge them.
+  #
+  # @return [Array<Hash>] possibly empty list of messages
+  def poll
+    create_subscription unless subscription_exists?
```

Comment:
how about hiding this logic behind a private method `find_or_create_subscription` which creates it unless it exists?